pub/htsget-openapi.yaml

```
@@ -309,16 +178,184 @@ components:
+          properties:
+            htsget:
+              properties:
+                error:
+                  example: PermissionDenied
+                message:
+                  example: Client is not authorized to access this resource
+    NotFoundError:
+      allOf:
+        - '$ref': '#/components/schemas/Error'
+        - type: object
+          description: The resource requested was not found
+    UnsupportedFormatError:
+      allOf:
+        - '$ref': '#/components/schemas/Error'
+        - type: object
+          description: The requested file format is not supported by the server
+          properties:
+            htsget:
+              properties:
+                error:
+                  example: UnsupportedFormat
+                message:
+                  example: The requested file format is not supported by the server
+    InvalidInputError:
+      allOf:
+        - '$ref': '#/components/schemas/Error'
+        - type: object
+          description: The request parameters do not adhere to the specification
+          properties:
+            htsget:
+              properties:
+                error:
+                  example: InvalidInput
+                message:
+                  example: The request parameters do not adhere to the specification
+    InvalidRangeError:
+      allOf:
+        - '$ref': '#/components/schemas/Error'
+        - type: object
+          description: The requested range cannot be satisfied
+          properties:
+            htsget:
+              properties:
+                error:
+                  example: InvalidRange
+                message:
+                  example: The requested range cannot be satisfied
+    InternalServerError:
+      allOf:
+        - '$ref': '#/components/schemas/Error'
+        - type: object
+          description: Internal server error
+          properties:
+            htsget:
+              properties:
+                error:
+                  example: InternalServerError
+                message:
+                  example: An internal server error occurred
+  parameters:
+    idParam:
+      in: "path"
+      name: id
+      description: read or variant object identifier
+      required: true
+      schema:
+        type: string
+    readsFormatParam:
+      in: "query"
+      name: format
+      description: File format, BAM (default), CRAM.
+      example: BAM
+      required: false
+      schema:
+        type: string
+        enum: [BAM, CRAM]
+    variantsFormatParam:
+      in: "query"
+      name: format
+      description: File format, VCF (default), BCF.
+      example: "VCF"
+      required: false
+      schema:
+        type: string
+        enum: [VCF, BCF]
+    classParam:
+      in: "query"
+      name: class
+      description: Request different classes of data (such as header or body).
+      example: "header"
+      required: false
+      schema:
+        type: string
+        enum: [header, body]
```

The only defined value for this field is `header`.

```suggestion
        enum: [header]
```